### DISCUSSION TITLE
Release 4.3.0 - React 18 Support

### DISCUSSION THREAD

**Discussion by danielbarion** [ReactTooltip/react-tooltip#797] [General]
Hi guys,

Someone with access into the NPM package, please:

1. Clone the master branch or `git pull`
2. run `yarn semantic-release`

I'll paste my console output as the semantic release is running in DryRun and I don't have permission to do a new release at NPM:
https://semantic-release.gitbook.io/semantic-release/usage/configuration#dryrun

## Release

# 4.3.0 (https://github.com/wwayne/react-tooltip/compare/v4.2.21...v4.3.0) (2022-10-12)

### Bug Fixes

    * #759 (https://github.com/wwayne/react-tooltip/issues/759) - Disable tooltip when disable prop changes (8e9cca0 (https://github.com/wwayne/react-tooltip/commit/8e9cca016f8eb59036c0c4623b85bf3f465ba262))
    * aftershow: call afterShow only after state has fully updated (54752e8 (https://github.com/wwayne/react-tooltip/commit/54752e81093f4c41bae1b49f7de65f28cf54df1a))
    * effect and type not properly applied at first render (a8d0e51 (https://github.com/wwayne/react-tooltip/commit/a8d0e51dd5dc0eb85d985232da90d54fd031ae4b))
    * getposition: no error when all placements outside (249b925 (https://github.com/wwayne/react-tooltip/commit/249b925a30ac2e7f04648a20bc9976e85358205b))
    * performance issue caused by excessive use of clearTimeout/Interval (22aea50 (https://github.com/wwayne/react-tooltip/commit/22aea5086db47727feae609bb7a6f8e269b7f2de))
    * react18: tip not hiding with React 18 StrictMode enabled (83b72c6 (https://github.com/wwayne/react-tooltip/commit/83b72c63ea9fe527bcbfb0b85414527e172c970b))
    * set aria-describedby value wrong when custom id (a04d26c (https://github.com/wwayne/react-tooltip/commit/a04d26cc0639054b474ca8b373c95a09b199973a))

### Features

    * component: adding "padding" property to customize padding style (9ae765a (https://github.com/wwayne/react-tooltip/commit/9ae765a3cc38b36ce5ea77edd8fcc0f7bd9646a6))
    * getposition: support desired-place-list (by priority) (982d89d (https://github.com/wwayne/react-tooltip/commit/982d89d2b9d663ed92d41743f63069d0c8a4aac0))
    * parent_body_mode: #535 (https://github.com/wwayne/react-tooltip/issues/535) search parents for data-tip if in body mode (71da348 (https://github.com/wwayne/react-tooltip/commit/71da34870163c4377a82b009ac63692e34eeb444))
    * pencil: readme updated to trigger next minor release (react 18 support release) (bb9c79e (https://github.com/wwayne/react-tooltip/commit/bb9c79e248918e4b0641b9305752e59019a60a72))
    * pencil: readme updated to trigger next minor release (react 18 support release) (3a6ec6c (https://github.com/wwayne/react-tooltip/commit/3a6ec6c1a1c61434532a9350e7ceafe17e8794b9))

**Comment by danielbarion** [comment]
cc @aronhelser @wwayne (that's very important guys)

**Reply by aronhelser** [reply]
I'm trying it....

**Comment by aronhelser** [comment]
Does the same for me - ` This run was not triggered in a known CI environment, running in dry-run mode.`
semantic-release is designed to be run automatically in CI - but I do not recall how it was set up. Potentially you need to either do `npm publish` command manually or figure out why sematic-release isn't running when a PR is merged.

I've invited you on NPM. @pdeszynski too.

**Reply by danielbarion** [reply]
Thanks Aron!

I'll check how to add that steps into the github actions and configure that one on the repository.

Oh, got it!

**Reply by aronhelser** [reply]
One thing I noticed is that npm lists the old repository address in wwayne's namespace (of course!) so I'm curious whether you will be able to publish the same package from a new repository address. Surely you must be able to do that....

**Reply by pdeszynski** [reply]
Hello all.

It should not matter from where you publish. You can do it from your own machine with --ci=false flag but you need npm token AND BE sure to run 'yarn build' earlier

You can check how I've setup GH actions on my fork https://github.com/pdeszynski/react-tooltip/tree/master/.github/workflows

**Reply by danielbarion** [reply]
Thanks, @pdeszynski!

It worked ☺️ 

I just did a release manually in the GitHub releases, I know that we can automate that, if you have time, can you take a look, please?

I'll try to take a look too when I have some available time

**Comment by danielbarion** [comment]
I'll lock that one as it is already solved, thank you, guys!
